app/com/linkedin/drelephant/analysis/AnalyticJobGeneratorHadoop2.java

```diff
@@ -122,40 +122,61 @@ public void configure(Configuration configuration)
    * @throws AuthenticationException
    */
   @Override
-  public List<AnalyticJob> fetchAnalyticJobs()
+  public List<AnalyticJob> fetchCompletedAnalyticJobs(long from, long to)
       throws IOException, AuthenticationException {
     List<AnalyticJob> appList = new ArrayList<AnalyticJob>();
-
-    // There is a lag of job data from AM/NM to JobHistoryServer HDFS, we shouldn't use the current time, since there
-    // might be new jobs arriving after we fetch jobs. We provide one minute delay to address this lag.
-    _currentTime = System.currentTimeMillis() - FETCH_DELAY;
-    updateAuthToken();
-
-    logger.info("Fetching recent finished application runs between last time: " + (_lastTime + 1)
-        + ", and current time: " + _currentTime);
+    logger.info("Fetching all the completed applications");
 
     // Fetch all succeeded apps
     URL succeededAppsURL = new URL(new URL("http://" + _resourceManagerAddress), String.format(
-            "/ws/v1/cluster/apps?finalStatus=SUCCEEDED&finishedTimeBegin=%s&finishedTimeEnd=%s",
-            String.valueOf(_lastTime + 1), String.valueOf(_currentTime)));
+            "/ws/v1/cluster/apps?finalStatus=SUCCEEDED&startedTimeBegin=%s&startedTimeEnd=%s",
```

Comment:
Earlier, when we were analyzing only succeeded and failed jobs, it made sense to use the finishedTime. But now that we analyze running jobs as well, we need to capture them soon after they start. For running jobs the finished time returned by the resource manager api is 0.

To answer your second question, this should not cause any problem because when we restart Dr. Elephant, we fetch all the jobs available in the resource manager starting from start time = 0(epoch) and then check in our database, if the application is already analyzed. If no, then we add it to the queue else we drop it.